### PR TITLE
SL-20635 - new data to ObjectUpdate and AvatarAppearance message

### PR DESCRIPTION
**Pull request by simon-linden**
The Data field usage is extended to have the total number of prims for a root prim.   AvatarAppearance has a new block listing attachment points and IDs.   See the jira for details

### CODE REVIEW

File: indra/newview/VIEWER_VERSION.txt [hunk 1]
```diff
@@ -1 +1 @@
-7.1.2
\ No newline at end of file
+7.1.2
```
Comment (simon-linden) [inline indra/newview/VIEWER_VERSION.txt:1]: This was failing the pre-commit hook checking for an EOF, so I opened and re-saved it

File: indra/newview/llviewerobject.cpp [hunk 2]
```diff
@@ -2751,22 +2643,6 @@ void LLViewerObject::interpolateLinearMotion(const F64SecondsImplicit& frame_tim
 
 
 
-BOOL LLViewerObject::setData(const U8 *datap, const U32 data_size)
```
Comment (simon-linden) [inline indra/newview/llviewerobject.cpp:2754]: Never used

File: indra/newview/llviewerobject.cpp [hunk 3]
```diff
@@ -1718,33 +1627,14 @@ U32 LLViewerObject::processUpdateMessage(LLMessageSystem *mesgsys,
 					setAngularVelocity(new_angv);
 					break;
 
-				case 16:
```
Comment (simon-linden) [inline indra/newview/llviewerobject.cpp:1721]: This format is never used by the simulator

File: indra/newview/llviewerobject.cpp [hunk 4]
```diff
@@ -1496,20 +1417,45 @@ U32 LLViewerObject::processUpdateMessage(LLMessageSystem *mesgsys,
...
-				// Check for appended generic data
-				S32 data_size = mesgsys->getSizeFast(_PREHASH_ObjectData, block_num, _PREHASH_Data);
-				if (data_size <= 0)
-				{
-					mData = NULL;
-				}
-				else
-				{
-					// ...has generic data
-					mData = new U8[data_size];
```
Comment (simon-linden) [inline indra/newview/llviewerobject.cpp:1510]: Older viewers with this code will end up allocating 1 byte for every linkset root object and never using the new data.   This has the potential to impact the viewer - we need to measure this and perhaps make it smarter to not send to old viewers.

**Comment by AtlasLinden** [comment]
Archived issue: https://github.com/secondlife/jira-archive-internal/issues/71011

**Pull request merged**